modules/data.atmosphere/R/met2CF.csv.R

```diff
@@ -5,245 +5,511 @@
+##'   OPTIONAL:
+##'   format$na.strings = list of missing values to convert to NA, such as -9999
+##'   format$skip = lines to skip excluding header
+##'   format$vars$column_number = Column number in CSV file (optional, will use header name first)
+##' Columns with NA for bety variable name are dropped. 
+##' Units for datetime field are the lubridate function that will be used to parse the date (e.g. \code{ymd_hms} or \code{mdy_hm}). 
 ##' @param nc_verbose logical: run ncvar_add in verbose mode?
 ##' @export
-##' @author Mike Dietze, David LeBauer
+##' @author Mike Dietze, David LeBauer, Ankur Desai
 ##' @examples
 ##' \dontrun{
-##'   in.path = "~/Downloads/"
-##'   in.file = "WR_E"
-##'   outfolder = "/tmp/"
-##'   format = list(
-##'                 orig_name = c("TA","PRECIP","RH","WS","WD","SW","PAR_in"),
-##'                 orig_units =  c("celsius","mm","%","m/s","degrees","W m-2","umol m-2 s-1"),
-##'                 bety_name = c("airT","precipitation_flux","relative_humidity","Wspd","wind_direction","solar_radiation","PAR"),
-##'                 CF_name = df$CF_name, 
-##'                 CF_units = df$CF_units,
-##'                 skip=7,
-##'                 header=1,
-##'                 na.strings=c("-9999","-6999","9999"))  
-##'   lat = 40
-##'   lon = -80
-##'   met2CF.csv(in.path,in.file,outfolder,format,lat,lon)
+##' library(PEcAn.DB)
+##' library(lubridate)
+##' library(RPostgreSQL)
+##' bety = list(user="bety", password="bety", host="localhost", dbname="bety", driver="PostgreSQL", write=TRUE)
+##' con <- db.open(bety)
+##' in.path <- '/home/carya/sites/willow/'
+##' in.prefix <- 'FLX_US-WCr_FLUXNET2015_SUBSET_HH_1999-2014_1-1'
+##' outfolder <- '~/'
+##' input.id <- 5000000005
+##' format <- query.format.vars(input.id=input.id,con)
+##' start_date <- ymd_hm('200401010000')
+##' end_date <- ymd_hm('200412312330')
+##' PEcAn.data.atmosphere::met2CF.csv(in.path,in.prefix,outfolder,start_date,end_date,format,overwrite=TRUE)
 ##' }
-met2CF.csv <- function(in.path, in.file, outfolder, format, lat=NULL, lon=NULL, nc_verbose = FALSE,...){
+met2CF.csv <- function(in.path, in.prefix, outfolder, start_date, end_date, format, lat=NULL, lon=NULL, nc_verbose = FALSE, overwrite=FALSE, ...){
+  require(PEcAn.utils)
+  require(lubridate)
+  require(udunits2)  
+  require(ncdf4)
   
+  start_year <- year(start_date)
+  end_year <- year(end_date)
+  if(!file.exists(outfolder)){
+    dir.create(outfolder)
+  }
+  
+  ## set up results output to return
+  rows <- end_year - start_year + 1
+  results <- data.frame(file=character(rows), host=character(rows),
+                        mimetype=character(rows), formatname=character(rows),
+                        startdate=character(rows), enddate=character(rows),
+                        dbfile.name = in.prefix,
+                        stringsAsFactors = FALSE)
+  
+  in.file <- paste0(in.prefix,'.csv')
```

Comment:
This seems like it could be both superfluous and error prone. Superfluous because two lines below we grep for files ending in csv. Error prone because if in.prefix already ends in csv (e.g. foo.csv), we're now going to search for foo.csv.csv and get no hits.